redisvl/query/filter.py

```diff
@@ -562,3 +565,394 @@ def __str__(self) -> str:
         if not self._filter:
             raise ValueError("Improperly initialized FilterExpression")
         return self._filter
+
+
+class Timestamp(FilterField):
+    """
+    A timestamp filter for querying date/time fields in Redis.
+
+    This filter can handle various date and time formats, including:
+    - datetime objects (with or without timezone)
+    - date objects
+    - ISO-8601 formatted strings
+    - Unix timestamps (as integers or floats)
+
+    All timestamps are converted to Unix timestamps in UTC for consistency.
+    """
+
+    OPERATORS = {
+        FilterOperator.EQ: "==",
+        FilterOperator.NE: "!=",
+        FilterOperator.LT: "<",
+        FilterOperator.GT: ">",
+        FilterOperator.LE: "<=",
+        FilterOperator.GE: ">=",
+        FilterOperator.BETWEEN: "between",
+    }
+
+    OPERATOR_MAP = {
+        FilterOperator.EQ: "@%s:[%s %s]",  # For exact timestamp match (converted to range for dates)
+        FilterOperator.NE: "(-@%s:[%s %s])",
+        FilterOperator.GT: "@%s:[(%s +inf]",
+        FilterOperator.LT: "@%s:[-inf (%s]",
+        FilterOperator.GE: "@%s:[%s +inf]",
+        FilterOperator.LE: "@%s:[-inf %s]",
+        FilterOperator.BETWEEN: "@%s:[(%s (%s]",  # should between be inclusive?
```

Comment:
Right now "between" is defined as non-inclusive - not sure if that's the right decision could see it either way.

Reply:
I suggest making this inclusive by default and either parameterize it or embed meaning into multiple operators. Since many people are familiar with Pandas, we might follow the Pandas interface, which uses an `inclusive` kwarg: https://pandas.pydata.org/docs/reference/api/pandas.date_range.html